### PR TITLE
Updating cirrus fingerprint script to include goldens version

### PR DESCRIPTION
**Pull request by Piinks**
## Description

Local golden file changes should not exist on Cirrus, but were introduced in the reverted PR #33688. This removes those local changes that are creating a sync error for the goldens repository on cirrus, and maintains a safe guard against further errant local changes, thereby preserving the testing environment.

The fingerprint script for cirrus has been updated to incorporate the goldens version when deciding to invalidate the cache or not. 

## Related Issues

This blocks #36087 and any further updates to the goldens repository.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_goldens_client/lib/client.dart [hunk 1]
```diff
@@ -119,13 +119,39 @@ class GoldensClient {
   }
 
   Future<void> _checkCanSync() async {
+    final bool isCirrus = (platform.environment['CIRRUS_CI'] ?? '').isNotEmpty;
```
Comment (dnfield) [inline packages/flutter_goldens_client/lib/client.dart]: nit:  I think we check this elsewhwer as just `platform.environment.containsKey('CIRRUS_CI')`

File: packages/flutter_goldens_client/lib/client.dart [hunk 2]
```diff
@@ -119,13 +119,39 @@ class GoldensClient {
...
     final io.ProcessResult result = await process.run(
       <String>['git', 'status', '--porcelain'],
       workingDirectory: repositoryRoot.path,
     );
+    print('Initial check: ${result.stdout}');
```
Comment (dnfield) [inline packages/flutter_goldens_client/lib/client.dart]: If you meant to leave this in, maybe we can give it a clearer label?

Comment (Piinks) [reply packages/flutter_goldens_client/lib/client.dart]: I forgot to add the WIP label. ;)

**Comment by Piinks** [comment]
cc/ @LongCatIsLooong

**Review by dnfield — APPROVED** [review]
LGTM

File: .cirrus.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@ task:
     fingerprint_script: echo $OS; grep -r --include=pubspec.yaml 'PUBSPEC CHECKSUM' "$CIRRUS_WORKING_DIR"
   flutter_pkg_cache:
     folder: bin/cache/pkg
-    fingerprint_script: echo $OS; cat bin/internal/engine.version
+    fingerprint_script: echo $OS; cat bin/internal/*.version
```
Comment (fkorotkov) [inline .cirrus.yml]: I think you missed it for other platforms like Windows and macOS

**Review by fkorotkov — CHANGES_REQUESTED** [review]

File: .cirrus.yml (hunk 3 above)
Comment (Piinks) [reply .cirrus.yml]: Ah good point. Goldens are currently only tested on Linux, but we should keep it consistent regardless. Thanks!

**Pull request merged**